### PR TITLE
fix how realm tests access proxy locations

### PR DESCRIPTION
**Pull request by pondrejk**
this was causing `BoxKeyError` at some of realm tests

### CODE REVIEW

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_realm.py -k test_negative_realm_update_invalid_type

File: tests/foreman/destructive/test_realm.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def test_positive_realm_info_name(
             'name': gen_string('alpha', random.randint(1, 30)),
             'realm-proxy-id': module_fake_proxy.id,
             'realm-type': 'Red Hat Identity Management',
-            'locations': [loc.read().name for loc in module_fake_proxy.location],
+            'locations': module_fake_proxy.locations.to_list(),
         }
     )
     request.addfinalizer(lambda: module_target_sat.cli.Realm(realm['id']).delete())
```
Comment (Gauravtalreja1) [inline tests/foreman/destructive/test_realm.py:94]: PRT failures appear to be caused by an incorrect fix in cleanup.py refactor [PR](https://github.com/SatelliteQE/robottelo/pull/9768/files#diff-d3671555df65a9a9c09455290c030fbe7f6a5ac589d62bbf77da561863cf9faaR94)
```suggestion
    request.addfinalizer(lambda: module_target_sat.cli.Realm.delete({'id': realm['id']}))
```

**Review by Gauravtalreja1 — CHANGES_REQUESTED** [review]

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_realm.py -k test_negative_realm_update_invalid_type

**Comment by pondrejk** [comment]
@Gauravtalreja1 thanks for that advice, passes now

**Review by Gauravtalreja1 — APPROVED** [review]

**Review by jameerpathan111 — APPROVED** [review]

**Pull request merged**